core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -75,11 +104,47 @@ class HttpSamDAO(baseSamServiceURL: String, serviceAccountCreds: Credential)(imp
     }
   }
 
-  override def getPolicySyncStatus(resourceTypeName: SamResourceTypeName, resourceId: String, policyName: SamResourcePolicyName, userInfo: UserInfo): Future[SamPolicySyncStatus] = {
-    val url = samServiceURL + s"/api/google/v1/resource/${resourceTypeName.value}/$resourceId/$policyName/sync"
+  private class SamApiCallback[T](functionName: String) extends ApiCallback[T] {
+    private val promise = Promise[T]()
```

Comment:
FWIW cats-effect `IO.async` has similar functionality to `Promise` where you can complete it from a callback… but Scala Promise/Future is fine here if that's the pattern in Rawls.

Reply:
I don't want to bite off switching this from Future to IO